src/core/main_task.c

```diff
@@ -63,6 +73,31 @@ void MainTask(void *pvParameters)
 {
     TickType_t xLastWakeTime = xTaskGetTickCount();
 
+    write_time = xTaskGetTickCount();
+
+    uint8_t eeprom_init_flag_addr = 0;
+    uint32_t operating_time_addr = 1;
+    uint32_t cycles_number_addr = 5;
```

Comment:
Addresses should be pointers:
```suggestion
    uint32_t *eeprom_init_flag_addr = (uint32_t *) 0;
    uint32_t *operating_time_addr = (uint32_t *) 4;
    uint32_t *cycles_number_addr = (uint32_t *) 8;
```